langroid/agent/batch.py

```diff
@@ -19,66 +22,161 @@
 
 load_dotenv()
 
+T = TypeVar("T")
+U = TypeVar("U")
 
-def run_batch_tasks(
-    task: Task,
-    items: List[Any],
-    input_map: Callable[[Any], str | ChatDocument] = lambda x: str(x),
-    output_map: Callable[[ChatDocument | None], Any] = lambda x: x,
+
+def run_batch_task_gen(
+    gen_task: Callable[[int], Task],
+    items: list[T],
+    input_map: Callable[[T], str | ChatDocument] = lambda x: str(x),
+    output_map: Callable[[ChatDocument | None], U] = lambda x: x,  # type: ignore
     sequential: bool = True,
+    batch_size: Optional[int] = None,
+    console_status: bool = True,
```

Comment:
I was thinking `quiet_mode` should take care of this. We run all batch tasks in quiet mode, so for example DocChat Agent should look at the global `settings.quiet` and decide whether to use console spinner, no?
If that works, this would obviate a lot of code changes related to this.

Reply:
For example in `chat_agent.py` I conditionally use rich spinner based on `settings.quiet`:

https://github.com/langroid/langroid/blob/main/langroid/agent/chat_agent.py#L684

```
        with ExitStack() as stack:  # for conditionally using rich spinner
            if not self.llm.get_stream() and not settings.quiet:
                # show rich spinner only if not streaming!
                cm = console.status("LLM responding to messages...")
                stack.enter_context(cm)
            if self.llm.get_stream() and not settings.quiet:
                console.print(f"[green]{self.indent}", end="")

```

Reply:
The spinner in `run_batch_tasks` could also cause conflicts, since it doesn't key off `quiet_mode` (for example, conflicts between this spinner and the spinner from `RelevanceExtractor` could cause a problem). But if all of these are controlled by `quiet_mode`, this would work (though it would probably be a more complex change).